src/dune/lib.ml

```diff
@@ -1871,10 +1874,16 @@ let to_dune_lib ({ info; _ } as lib) ~modules ~foreign_objects ~dir =
   let info = Lib_info.set_ppx_runtime_deps info ppx_runtime_deps in
   let info = Lib_info.set_sub_systems info (Sub_system.public_info lib) in
   let* main_module_name = main_module_name lib in
-  let+ requires = lib.requires in
+  let* requires = lib.requires in
   let requires = add_loc requires in
-  Dune_package.Lib.make ~info ~requires ~modules:(Some modules)
-    ~main_module_name
+  let+ re_exports = lib.re_exports in
+  let re_exports = List.map ~f:(fun t -> (Loc.none, t.name)) re_exports in
+  let requires =
+    List.map ~f:Lib_dep.direct requires
+    @ List.map ~f:Lib_dep.re_export re_exports
+  in
+  let info = Lib_info.set_requires info requires in
```

Comment:
These successive `Lib_info.set_xxx` feels inefficient BTW, we should have a single `Lib.set` function that takes multiple fields to avoid creating many intermediate records.

Reply:
Yeah, good call. I will turn this into a single call.